ntbs-integration-tests/NotificationPages/ManualTestResultEditPagesTests.cs

```diff
@@ -210,10 +213,58 @@ public async Task PostDelete_ReturnsSuccessAndRemovesResult()
             var result = await SendPostFormWithData(editDocument, formData, editUrl, "Delete");
 
             // Assert
-            result.AssertRedirectTo(GetPathForId(NotificationSubPaths.EditTestResults, NotificationId));
+            result.AssertRedirectTo(GetPathForId(NotificationSubPaths.EditTestResults, notificationId));
             var testsListPage = await Client.GetAsync(GetRedirectLocation(result)); // Follow the redirect to see results table
             var testsListDocument = await GetDocumentAsync(testsListPage);
             Assert.Null(testsListDocument.GetElementById($"manual-test-result-{TEST_TO_DELETE_ID}"));
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task GetFilteredSampleTypesForManualTestType_ReturnsExpectedValues(bool isEdit)
+        {
+            // Arrange
+            const int manualTestTypeId = (int)ManualTestTypeId.Histology;
+            const string handlerPath = "FilteredSampleTypesForManualTestType";
+            var endpointPath = $"{NotificationSubPath}{(isEdit ? "0/" : "")}{handlerPath}";
+            var endpointUrl = GetPathForId($"{endpointPath}?value={manualTestTypeId.ToString()}", 0);
+
+            // Act
+            var response = await Client.GetAsync(endpointUrl);
+
+            // Assert
+            var result = await response.Content.ReadAsStringAsync();
+            var filteredLists = JsonConvert.DeserializeObject<FilteredManualTestPageSelectLists>(result);
+            Assert.Equal(((int)SampleTypeId.BronchoscopySample).ToString(), filteredLists.SampleTypes.First().Value);
+            Assert.Equal("BRONCHOSCOPY SAMPLE", filteredLists.SampleTypes.First().Text.ToUpperInvariant());
+            Assert.Equal("RESPIRATORY", filteredLists.SampleTypes.First().Group.ToUpperInvariant());
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task IfDateTooEarly_ValidateTestResultForEditDate_ReturnsEarliestDateErrorMessage(bool isEdit)
+        {
+            // Arrange
+            var formData = new Dictionary<string, string>
+            {
+                ["key"] = "TestDate",
+                ["day"] = "1",
+                ["month"] = "1",
+                ["year"] = "1899"
+            };
+            const string handlerPath = "ValidateTestResultForEditDate";
+            var queryString = string.Join("&", formData.Select(kvp => $"{kvp.Key}={kvp.Value}"));
+            var endpointPath = $"{NotificationSubPath}{(isEdit ? "0/" : "")}{handlerPath}";
```

Comment:
Why is the Id 0 for the edit case, shouldn't it be the actual id of the manual test type? Also, does this path actually resolve for the case where no id is provided?
I'm working on top off this branch 3 commits behind, and in my checkout the dynamic validation for a NEW manual test result (i.e. where there is no id in the path) does not work as the path doesn't get resolved - if it works for you, do you know what fixed it? I'm doing the same thing for social context venues.

Reply:
You've already spotted the split between edit/add. 
Id = 0, as we don't actually care what RowId is, we're just hitting the dynamic endpoints accessed normally through components.

Reply:
Yep, I realised the actual value of the Id didn't matter after I wrote the comment